lib/zendesk_apps_support/validations/requests.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'ipaddress'
+
+module IPAddress
+  class IPv4
+    # TODO: remove when/if a new version of https://github.com/ipaddress-gem/ipaddress is released
+    def link_local?
+      [self.class.new('169.254.0.0/16')].any? { |i| i.include? self }
+    end
+  end
+end
+
+module ZendeskAppsSupport
+  module Validations
+    module Requests
+      class << self
+        HTTP_REQUEST_METHODS = %w[GET HEAD POST PUT DELETE CONNECT OPTIONS TRACE].freeze
+
+        REQUEST_CALLS = [
+          /\w+\.request\(['"](.*)['"]/i, # ZAF request
+          /\$\.(?:get|post|getJSON)\(['"](.*)['"]/i, # jQuery request
+          /\w+\.open\(['"](?:#{Regexp.union(HTTP_REQUEST_METHODS).source})['"],\s?['"](.*)['"]/i, # XMLHttpRequest
+          /fetch\(['"](.*)['"]/i # fetch
+        ].freeze
+
+        def call(package)
+          errors = []
+          files = package.js_files + package.html_files
+
+          files.each do |file|
+            contents = file.read
+            REQUEST_CALLS.each do |request_pattern|
+              request = contents.match(request_pattern)
+              next unless request
+              uri = URI(request.captures[0])
+              package.warnings << "Possible insecure HTTP request to #{uri} in #{file.relative_path}. Consider instead using the HTTPS protocol." if uri.scheme == 'http'
+
+              next unless IPAddress.valid? uri.host
+              ip = IPAddress uri.host
```

Comment:
What's going on here, syntactically? Never seen calling a class like a function before.

Reply:
...good question. I think it's defaulting to the class `parse` method, though I'm not sure how exactly that works. I can look into this

Reply:
Ah, I see: https://github.com/ipaddress-gem/ipaddress/blob/a8eb06d418efc1b3bb036c849c95cf0ce4b7e18b/lib/ipaddress.rb#L276

TBH I have a slight preference for just `IPAddress::parse` - this kind of syntactic sugar makes my skin crawl.

Reply:
Yep that's fair - I'll update this